修改多处

stevending1st opened the pull request:
之前匆匆瞥过数眼，未细看，前几日于 B 站看林旅强先生之解说，今日抽空细读，修改数处，不当之处请多包涵。
详情请参见每次 commit。

stevending1st committed:
add a Space

add a Space

stevending1st committed:
change the punctuation

逗号和顿号都表示停顿，但逗号表示的停顿长，顿号表示的停顿短。并列词语之间的停顿一般用顿号，但当并列词语较长或其后有语气词时，为了表示稍长一点的停顿，也可用逗号。

stevending1st committed:
关联词修改

stevending1st committed:
成语替换，添加关联词

敝帚自珍： 比喻东西虽然不好，自己却很珍惜。
秘而不露： 严守秘密，不肯吐露。
秘而不露相对恰当一点点。

stevending1st committed:
Uniform case

stevending1st committed:
修改关联词

stevending1st committed:
标点修改

两句有明显并列关系，可合为一句话。

stevending1st committed:
开放 小节句式调整

stevending1st committed:
平等小节修改

stevending1st committed:
删去“一些”，显得多余

stevending1st committed:
修改“协作”小节

stevending1st committed:
“创造美好世界” 一节修改

stevending1st committed:
修改“推而广之”小节

stevending1st committed:
修改“互帮互助”

stevending1st committed:
修改“立即行动”，“结语” 两小节

zhuangbiaowei commented:
@stevending1st 非常感谢您的PR。有很多修改我都非常赞同，有一些疑惑点，想跟你再讨论一下。

zhuangbiaowei commented on README.md:
```diff
@@ -4,46 +4,46 @@
 * [English version](README_en.md)
 
-我们是一群开源的爱好者与信仰者，我们相信：开源代表着一种向善的力量！作为一场席卷全球的世界性运动，20多年来的历史证明，开源不仅仅能够孕育最新的技术、创造更好的软件，更能够帮助这个世界变得更好。
+我们是一群开源的爱好者与信仰者，我们相信：开源代表着一种向善的力量！作为一场席卷全球的世界性运动，20 多年来的历史证明，开源不仅能够孕育最新的技术，创造更好的软件，而且能够帮助这个世界变得更好。
```
从：更能够帮助这个世界变得更好。
到：而且能够帮助这个世界变得更好。

我还是希望保留“更”，以体现递进的含义。

stevending1st replied:
原句：

> ...开源**不仅仅**能够孕育最新的技术、创造更好的软件，**更**能够帮助这个世界变得更好。

这里使用了关联词 `...... 不仅仅 ...... 更 ......`，目前似乎没有这种用法，
进而将 `更` 修改为 `而且`，使用了使用 `...... 不仅 ...... 而且 ......` 句型。
窃以为，`...... 不仅 ...... 而且 ......` 也存在递进关系，
如果老师执意要保留 `更`，以更明确递进关系，也可以修改成如下内容：

> ...开源能够孕育最新的技术、创造**更**好的软件，**更**能够帮助这个世界变得更好。

(这种方法值得注意的是两个**更**字非常贴近，朗读起来不够优美。

zhuangbiaowei replied:
是的，我想明白了，改成你的这个句子，更加合理。

zhuangbiaowei commented on README.md:
```diff
@@ -4,46 +4,46 @@
 * **分享（Sharing）**
 
-当一个软件工程师写出一个不错的软件，他不会敝帚自珍，不会故步自封。他乐于分享，是因为他相信：这个软件可能会对别人也有帮助，更会有人帮助他，一起做出更好的软件。西谚有云：赠人玫瑰，手留余香。我们都相信：**乐于分享是一切善举的开端。**
+当一个软件工程师写出一个不错的软件，他不会秘而不露，也不会故步自封。他乐于分享，是因为他相信：这个软件可能会对别人也有帮助，更会有人帮助他，一起做出更好的软件。西方谚语有云：“赠人玫瑰，手留余香”。我们都相信：**乐于分享是一切善举的开端。**
```
我觉得“敝帚自珍”，会比“秘而不露”更好一些。

stevending1st replied:
下面我先贴一下两个词的含义：
敝帚自珍： 比喻东西虽然不好，自己却很珍惜。
秘而不露： 严守秘密，不肯吐露。

在我的理解中，在该语境下，这个词想说明的是软件工程师不会把自己的作品藏起来，而是把它分享出来。
如果这个理解正确，那么这里的**敝帚自珍**属于成语使用不当了。

其实，我也筛选了很多成语，目前也就发现**秘而不露**相对恰当一点点。

zhuangbiaowei replied:
一般软件工程师，不愿意开源自己的代码，有两个原因。一种是觉得自己的代码写得太烂，不好意思见人。另一种是觉得自己的代码写得非常有“价值”，怕被人拿去了。我想指出的是后面那种心态，通常这种自以为写得很好的代码，其实并没有作者想象的那么好。所以，我称之为“敝帚自珍”。

stevending1st replied:
> 我想指出的是后面那种心态，通常这种自以为写得很好的代码，其实并没有作者想象的那么好。

敝帚自珍的含义——比喻东西虽然不好，自己却很珍惜。和您所指的不太一样，甚至是前后矛盾的。

stevending1st commented:
> 想跟你再讨论一下。

老师请讲。

zhuangbiaowei commented on README.md:
```diff
@@ -4,46 +4,46 @@
+* **开放（Openness）**
 
-在很多方面，开放都非常重要。不仅仅是开放源代码，更包括公开透明的社区。这样的社区能够吸引更多的朋友加入。也能够帮助新来者，理解并认同社区规则。还能够促进监督以提升社区运行的程序正义。开放还包括欢迎一切的可能性，开源是世界的，也欢迎来自世界任何一个角落的使用者、参与者和贡献者。中国谚语有云：海纳百川，有容乃大。我们都相信：**公开透明是一切良好协作的基石。**
+在很多方面，开放都非常重要。开放是开放源代码；开放也是建立公开透明的社区——这样的社区能够吸引更多的朋友加入，也能够帮助新来者理解并认同社区规则，还能够促进监督以提升社区运行的程序正义；开放也是欢迎一切的可能性，欢迎来自世界任何一个角落的使用者、参与者和贡献者。中国古语有云：“海纳百川，有容乃大”。我们都相信：**公开透明是一切良好协作的基石。**
```
我觉得“不仅仅是开放源代码，更包括公开透明的社区。”，比“开放是开放源代码；开放也是建立公开透明的社区”，要通顺一些。

stevending1st replied:
原句：

> ......不仅仅是开放源代码，更包括公开透明的社区......开放还包括欢迎一切的可能性.....

这里指出了“开放"的三道含义：开放源代码、公开透明的社区、欢迎一切的可能性

修改后：

> ...开放是开放源代码；开放也是建立公开透明的社区......开放也是欢迎一切的可能性......

句式统一。

和前面一样关联词没有`......不仅仅......更......`这种用法。

zhuangbiaowei replied:
这个...我还是觉得原来的描述更加顺畅。因为，我并不是在严格的定义开放的三层含义。而是在举例说明，为何开放非常重要。不仅仅开放源代码很重要，公开透明的社区也很重要。除了这些有形的开放，欢迎一切可能性，是更加彻底的开放。

stevending1st replied:
“重构”了一下：

> 开放的重要性体现在多个方面。开放源代码，可以让智慧跨越时间和空间在一起交融。建立公开透明的社区，能够吸引更多的朋友加入，也能够帮助新来者理解并认同社区规则，还能够提升监督水平以促进社区运行的程序正义。开放包括欢迎一切的可能性，欢迎来自世界任何一个角落的使用者、参与者和贡献者。中国古语有云：“海纳百川，有容乃大”。我们都相信：**公开透明是一切良好协作的基石。**

zhuangbiaowei replied:
这样很棒！

zhuangbiaowei commented on README.md:
```diff
@@ -4,46 +4,46 @@
 * **推而广之（Advocate widely）**
 
-我们应该更加努力的向大众传播开源的理念与精神，让更多的人接受开源的理念，成为开源的同道中人。我们还应该在开源软件、开源硬件之外的领域，推广开源的实践，不仅仅是开放源代码，还应该开放数据、开放知识、开放一切可以帮助这个世界变得更好的知识与经验，让更多的行业、更多的群体，都接纳开源，成为开放式协作的受益者。
+我们应该更加努力的向大众传播开源的理念与精神，让更多的人接受开源的理念，成为开源的同道中人。我们还应该在开源软件、开源硬件之外的领域，推广开源的实践——不仅开放源代码，而且开放数据，开放知识，开放经验，开放一切可以帮助这个世界变得更好的事物，让更多的行业、更多的群体都接纳开源，成为开放式协作的受益者。
```
“还应该开放数据”，我觉得这样的强调要好一些。另外，“知识与经验”，替换成“事物”，可能太过宽泛了。

stevending1st replied:
原句：

> ......我们还应该在开源**软件**、开源**硬件**之外的领域......不仅仅是开放**源代码**，还应该开放**数据**、开放**知识**、开放一切可以帮助这个世界变得更好的**知识与经验**......

这里提出了开放的内容：源代码、数据、知识、知识与经验。
很明显，“知识”使用了两次。
“知识与经验” 对 “源代码”、“数据”、“知识” 也不形成包含关系。

这里我也想了词，如“物质与意识”等，都不太满意，最后选定了“事物”，“事”为无形，“物”为有形，囊括了有形和无形即囊括了**一切**。
想必老师也是希望开源是**全领域的**，**一切**的吧。

zhuangbiaowei replied:
知识和经验，与前面的知识重复了。这个的确是问题，不过“事物”的确也太过宽泛了。我希望表达的是某种：可以被无限复制的无形的事物，这样的开放与分享，才是不受限制的。你看看，哪个词会更加好些？

stevending1st replied:
目前没有想到比较好的词，既然这样就去掉前一个“开放知识”。

> 推广开源的实践——不仅开放源代码，而且开放数据，开放一切可以帮助这个世界变得更好的知识与经验......

zhuangbiaowei replied:
这样更好 ！

zhuangbiaowei commented on README.md:
```diff
@@ -4,46 +4,46 @@
 * **互帮互助（Help each other）**
 
-我们应该帮助更多的开源项目，不断发展成长。帮助各个开源社区，把社区的力量团结起来，共同协作。我们还应该防止开源的含义被滥用或曲解。我们要阻止割裂，反对人为设置的障碍，反对任何附加歧视条款的“伪开源”，确保开源始终是一项惠及全球的事业。
+我们应该帮助更多开源项目，不断发展成长；帮助更多开源社区，团结力量，共同协作。我们应该防止开源的含义被滥用或曲解。我们应该阻止割裂，反对人为设置的障碍，反对任何附加歧视条款的“伪开源”，确保开源始终是一项惠及全球的事业。
```
把社区的力量团结起来，比单纯将“团结力量”，要更好一些。

stevending1st replied:
这里是为了句式的一致性，进行了调整。

zhuangbiaowei replied:
句式一致性，的确需要考虑。

帮助更多开源社区，团结互助，共同协作。

这样是不是更好些？

stevending1st replied:
开始我没理解老师的意思，下面我再改下：

> 我们应该帮助更多的开源项目，茁壮成长，臻于至善；帮助更多的开源社区，团结互助，共同协作。

zhuangbiaowei reviewed (commented):
> > 想跟你再讨论一下。
> 
> 老师请讲。

具体的文字，我都以code review的形式评论了。

stevending1st commented:
> > > 想跟你再讨论一下。
> > 
> > 
> > 老师请讲。
> 
> 具体的文字，我都以code review的形式评论了。

老师您好，我已经回复了，请指正。

zhuangbiaowei commented:
> > > > 想跟你再讨论一下。
> > > 
> > > 
> > > 老师请讲。
> > 
> > 具体的文字，我都以code review的形式评论了。
> 
> 老师您好，我已经回复了，请指正。
> 

非常感谢，我再仔细想想。

stevending1st committed:
一些修改

“重构”开放（Openness）一小节。
推而广之（Advocate widely）小节中去掉“开放知识”。
互帮互助（Help each other）小节中修改了第一句话，统一结构。

zhuangbiaowei commented:
@stevending1st 因为没收到你的后续回复，我打算先合入你的修改，在把我的一些改动，直接附加上去。后续如果有建议，也欢迎你继续提交PR。

感谢！

stevending1st commented:
> 这样修改如何：
> 从：
>  > 当一个软件工程师写出一个不错的软件，他不会秘而不露，也不会故步自封。他乐于分享，是因为他相信...
>
> 改成：
>
> > 当一个软件工程师写出一个不错的软件，他不会吝于分享，也不会怯于分享。他乐于分享，是因为他相信...

这个确实要好很多！

---

> @stevending1st 因为没收到你的后续回复，我打算先合入你的修改，在把我的一些改动，直接附加上去。后续如果有建议，也欢迎你继续提交PR。

抱歉，最近有些别的事，加上确实也没有特别好的方法，所以没有后续跟进。不过，目前的您提出的方案已经特别好了！

---

> 感谢！

不客气！
